07-javascript.js

```diff
@@ -143,6 +163,25 @@ assert(p.hasOwnProperty("name"))
 // Penalty: -3 layout, -3 naming
 
 //START
+function bugs(callback) {
+
+  callback = function(){
```

Comment:
Callback is the parameter passed in to the `bugs` function. That's what you are supposed to be calling once you've changed the meaning of `sup()`.

You've overwritten it here, so it will never get called.